channeldb/payment_control.go

```diff
@@ -463,7 +526,9 @@ func (p *PaymentControl) FetchInFlightPayments() ([]*InFlightPayment, error) {
 				return nil
 			}
 
-			inFlight := &InFlightPayment{}
+			inFlight := &InFlightPayment{
+				Attempts: make([]HTLCAttemptInfo, 0),
```

Comment:
It isn't the same. Removing this initializes `Attempts` as nil instead of an empty slice. It may introduce a subtle bug if some code checks for `nil` and there would also be another source of attempts that uses the empty slice. Generally I try to keep the contract as strict as possible. But removed to keep this pr moving.